controllers/storagecluster/onboarding_validation_keys_generator_job.go

```diff
@@ -0,0 +1,103 @@
+package storagecluster
+
+import (
+	"fmt"
+	"os"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+)
+
+const (
+	onboardingValidationKeysGeneratorImage   = "ONBOARDING_VALIDATION_KEYS_GENERATOR_IMAGE"
+	onboardingValidationKeysGeneratorJobName = "onboarding-validation-keys-generator"
+	onboardingValidationPublicKeySecretName  = "onboarding-ticket-key"
+)
+
+type onboardingValidationKeysGeneratorJob struct{}
+
+var _ resourceManager = &onboardingValidationKeysGeneratorJob{}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if instance.Spec.AllowRemoteStorageConsumers {
+		var err error
+		r.Log.Info("onboarding validation key generator job: Spec.AllowRemoteStorageConsumers is enabled")
```

Comment:
Could we reformat the message to something like:
```suggestion
		r.Log.Info("Spec.AllowRemoteStorageConsumers is enabled. Creating Onboarding validation key generator job")
```